### DISCUSSION TITLE
RPM dependency on erlang

### DISCUSSION THREAD

**Discussion by maederm** [rabbitmq/rabbitmq-server#5079] [Questions] [answered]
I have a question about using dnf to install rabbitmq. It seems that the dependencies on the rabbitmq-server only specifies erlang >= 23.2. However in the erlang repo there is already an 25.x release which isn't supported for that version of rabbitmq-server according to [1]. With the last system upgrade we run into an issue with an incompatible erlang release.
Do I have to versionlock the package? Naively I assumed that the rabbitmq-server package would specify the allowed version range of erlang.
[1] https://www.rabbitmq.com/which-erlang.html

**Comment by michaelklishin** [comment, marked as answer]
Erlang 25 is supported by RabbitMQ 3.10 and latest 3.9 and 3.8 patch releases run on it. You should be able to pin Erlang to a specific version if you really have to.

**Reply by maederm** [reply]
As an example, if you install rabbitmq-server with [release 3.9.20](https://cloudsmith.io/~rabbitmq/repos/rabbitmq-server/packages/detail/rpm/rabbitmq-server/3.9.20-1.el8/a=noarch;d=el%252F8;t=binary/) on RHEL8, you would get an incompatible erlang release. Is that intended?

To be more clear, shouldn't the package rabbitmq-server next to erlang >= 0:23.2 also have the dependency erlang <= 0:24.3?

**Reply by michaelklishin** [reply]
You do get a compatible version, we just don't list Erlang 25 as supported in the docs so that people upgrade to 3.10 at the same time.

You can pin [Erlang to a specific version](https://access.redhat.com/solutions/98873).

**Reply by maederm** [reply]
Fair enough, we ran into issues because of that behavior (we basically pinned Rabbitmq without the Erlang release). 

`rabbitmq-plugins` crashed with `please re-compile this module with an Erlang/OTP 25 compiler`. Rabbitmq itself kept running.

**Reply by michaelklishin** [reply]
This is a known CLI tools behavior that's been addressed in `3.9.19` or so.
